src/config.rs

```diff
@@ -128,6 +150,29 @@ impl Context {
         self.get_config_int(key) != 0
     }
 
+    /// Gets configured "delete_server_after" value.
+    ///
+    /// `None` means never delete the message, `Some(0)` means delete
+    /// at once, `Some(x)` means delete after `x` seconds.
+    pub fn get_config_delete_server_after(&self) -> Option<i64> {
+        match self.get_config_int(Config::DeleteServerAfter) {
+            0 => None,
+            1 => Some(0),
+            x => Some(x as i64),
```

Comment:
get_config_int is an i32.  The setting of this afaik comes from the ffi as a string with no validation on it's value?
So while I think this cast itself is safe, the underlying conversions through the db are not.
Should we have a branch in set_config() which enforces either i32 or i64 (or even probably unsigned even) and correctly reverse here?  An additional privacy issue is that get_config_int silently converts to 0, i.e. never delete, on any problem like when it overflows i32.

Reply:
As an aside here, this kind of bug wouldn't exist on my longstanding which to strongly-type all values stored in the db.  This is not of use here, but to remember next time i try and introduce that as a point in favour ;)